packages/syft/src/syft/core/adp/data_subject_ledger.py

```diff
@@ -61,7 +61,10 @@ def convert_constants_to_indices(rdp_constant_array: np.ndarray) -> np.ndarray:
     """
     # Find indices for all RDP constants <= 50
     sub50_mask = rdp_constant_array <= 50
-    sub50_indices = (((rdp_constant_array - 1) * sub50_mask) * 10_000).astype(int)
+    # np.maximum is to avoid negative indices when rdp_constant_array is < 1
+    sub50_indices = (
+        (np.maximum(0, rdp_constant_array - 1) * sub50_mask) * 10_000
+    ).astype(int)
```

Comment:
MAYBE: Could we do 
```py
sub50_mask = rdp_constant_array <= 50
sub50_indices = ((rdp_constant_array * sub50_mask) * 10_000).astype(int)
```
by removing the minus -1 
maybe that would have caused the negative indices?

Reply:
The minus 1 was definitely causing the negative indices, but it is necessary to get the correct indexing (see the image below)
![image](https://user-images.githubusercontent.com/32711264/202245354-2b3f8476-17f9-4a60-b7f8-5dca65372a7e.png)

If you look at the notebook here, it has more details: https://github.com/OpenMined/PySyft/blob/dev/notebooks/Experimental/Ishan/ExtendRDPCache.ipynb

Reply:
Ahh , correct, 👍